### PR TITLE
[INLONG-4987][Agent] Support CloudEvents Source

### LINKED ISSUE #4987: [Feature][Agent] Support CloudEvents Source
### Description

CloudEvents is a specification for describing event data in common formats to provide interoperability across services, platforms and systems.

https://github.com/cloudevents/spec

### Use case

Kafka Example:
https://github.com/cloudevents/sdk-java/tree/master/examples/kafka

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by iamsee123**
### Prepare a Pull Request

- Fixes #4987

### Motivation

CloudEvents is a specification for describing event data in common formats to provide interoperability across services, platforms and systems.

### Modifications

+ add an util class to convert data with cloudevent format into default message

### Verifying this change

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [x] This change added tests and can be verified as follows:

 - TestCloudEventsConverterUtil: to test convert util with kafka

### CODE REVIEW

**Comment by GanfengTan** [comment]
Thanks your PR @iamsee123 . Is it more appropriate as an SDK？

**Comment by iamsee123** [comment]
> Thanks your PR @iamsee123 . Is it more appropriate as an SDK？

I think using util class is enough, because it just use to convert data from cloudEvent format into defaultMessage format.

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestCloudEventsConverterUtil.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ */
+public class TestCloudEventsConverterUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TestCloudEventsConverterUtil.class);
+
+    @Before
+    public void createKafkaProducer() {
+        //Basic producer configuration
+        Properties props = new Properties();
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "172.17.0.3:9092");
```
Comment (haibo-duan) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestCloudEventsConverterUtil.java:62]: Please change the IP to 127.0.0.1, and not use a real host.

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestCloudEventsConverterUtil.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        producer.close();
+    }
+
+    /**
+     * Just using in local test
+     */
+    @Ignore
+    public void testConverter() {
+        Properties props = new Properties();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "172.17.0.3:9092");
```
Comment (haibo-duan) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestCloudEventsConverterUtil.java:105]: Please change the IP to 127.0.0.1, and not use a real host.

**Comment by xuesongxs** [comment]
1、The implementation of this PR is a little different from my imagination. My goal is that Inlong starts an HTTP service, receives CloudEvents data, writes it to the Pulsar Topic, and the structure stored in the Topic is CloudEvents.
2、If you want to use the InlongMsg structure uniformly, use cloudevents as the body of InlongMsg inside Inlong.
3、CloudEvents can be transmitted based on HTTP, TCP and gPRC protocols.

**Comment by iamsee123** [comment]
> 1、The implementation of this PR is a little different from my imagination. My goal is that Inlong starts an HTTP service, receives CloudEvents data, writes it to the Pulsar Topic, and the structure stored in the Topic is CloudEvents. 2、If you want to use the InlongMsg structure uniformly, use cloudevents as the body of InlongMsg inside Inlong. 3、CloudEvents can be transmitted based on HTTP, TCP and gPRC protocols.

I got it. I will reconstruct the code later.

**Comment by xuesongxs** [comment]
The inlong full link must support CloudEvents protocol.
You can join the WeChat group to discuss.


 
From: iamsee123
Date: 2022-10-15 22:46
To: apache/inlong
CC: xuesongxs; Comment
Subject: Re: [apache/inlong] [INLONG-4987][Agent] Support CloudEvents Source (PR #5783)
1、The implementation of this PR is a little different from my imagination. My goal is that Inlong starts an HTTP service, receives CloudEvents data, writes it to the Pulsar Topic, and the structure stored in the Topic is CloudEvents. 2、If you want to use the InlongMsg structure uniformly, use cloudevents as the body of InlongMsg inside Inlong. 3、CloudEvents can be transmitted based on HTTP, TCP and gPRC protocols.
I got it. I will reconstruct the code later.

**Review by yunqingmoswu — APPROVED** [review]

**Comment by lucaspeng12138** [comment]
[自动回复]：您的邮件我已收到，我将尽快给您回复，谢谢

**Pull request closed without merging**